tempesta_fw/avx_routines.c

```diff
@@ -0,0 +1,58 @@
+
+const unsigned char __c_avx_data[64] __attribute__((aligned(64))) =
+{
+    //header charset
+    0xa8, 0xf8, 0xf8, 0xf8, 0xf8, 0xf8, 0xf8, 0xf8,
+    0xf8, 0xf8, 0xf0, 0x50, 0x50, 0x54, 0x50, 0x50,
+    //constants1
+    'A','A','A','A','Z','Z','Z','Z',
+    ' ',' ',' ',' ',':',':',':',':',
+    //constants2
+    0xF,0xF,0xF,0xF,0xF,0xF,0xF,0xF,
+    1,  2,  4,  8,  16, 32, 64, 128,
+    0,  0,  0,  0,  0,  0,  0,  0,
+    0,  0,  0,  0,  0,  0,  0,  0
+};
+
+const unsigned char __c_avx_header_names[4*64] __attribute__((aligned(64))) =
+{
+    0xFF,0xFF,0xFF,0xFF,0xFF,0x00,0x00,0x00,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+
+    'h','o','s','t', 0 , 0 , 0 , 0 ,
+    't','r','a','n','s','f','e','r',
+    'x','-','f','o','r','w','a','r',
+    'c','o','n','t','e','n','t','-',
+
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0x00,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0x00,
+
+    'c','a','c','h','e','-','c','o',
+    'c','o','o','k','i','e', 0,  0 ,
+    'c','o','n','n','e','c','t','i',
+    'u','s','e','r','-','a','g','e',
+
+    0xFF,0xFF,0xFF,0xFF,0xFF,0x00,0x00,0x00,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,
+    0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0xFF,0x00,
+
+    't','y','p','e', 0 , 0 , 0 , 0 ,
+    '-','e','n','c','o','d','i','n',
+    'd','e','d','-','f','o','r', 0 ,
+    'l','e','n','g','t','h', 0 , 0 ,
+
+    0xFF,0xFF,0xFF,0xFF,0xFF,0x00,0x00,0x00,
+    0xFF,0xFF,0xFF,0x00,0x00,0x00,0x00,0x00,
+    0xFF,0xFF,0xFF,0x00,0x00,0x00,0x00,0x00,
+    0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
+
+    'n','t','r','o','l', 0 , 0 , 0 ,
+    'g', 0 , 0 , 0 , 0 , 0 , 0 , 0 ,
+    'o','n', 0 , 0 , 0 , 0 , 0 , 0 ,
+    'n','t', 0 , 0 , 0 , 0 , 0 , 0 ,
+};
```

Comment:
The file contains constants only. And the constants are used in different compilation unit. Why did you move them to separate .c file? Moreover, since this is standalone translation unit, compiler can not efficiently optimize code using the constants and directly build them into code.